.github/workflows/main-push-build-test-and-publish-release.yaml

```diff
@@ -21,20 +21,26 @@ on:
       - main
 
 jobs:
-  main-push-build-test-and-publish-release:
+  main-create-timestamp-and-tag:
     runs-on: ubuntu-latest
+    outputs:
+      timestamp: ${{ steps.create_timestamp.outputs.timestamp }}
     steps:
       - name: "Checkout source code of the repository"
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: "Create timestamp"
+        id: create_timestamp
+        run: printf "timestamp=%s\n" "$(date +%Y%m%d-%H%M%S)" >> "$GITHUB_OUTPUT"
+
       - name: "Set Git identity"
         run: |
           git config --global user.email "github-actions[bot]@users.noreply.github.com"
           git config --global user.name "github-actions[bot]"
 
-      - name: "Set new version tag"
+      - name: "Create a new tag"
```

Comment:
I don't think it's good idea to create a tag for each commit in `main`

Reply:
This was introduced in https://github.com/dynatrace-oss/eBPF-Discovery/pull/101 by @lnarolski, I think that we shouldn't change it.